packages/devtools_app/lib/src/code_size/code_size_table.dart

```diff
@@ -1,73 +1,54 @@
 import 'package:flutter/material.dart';
-import 'package:provider/provider.dart';
 
-import '../auto_dispose_mixin.dart';
 import '../charts/treemap.dart';
 import '../table.dart';
 import '../table_data.dart';
 import '../utils.dart';
-import 'code_size_controller.dart';
 
-class TreemapTreeTable extends StatefulWidget {
-  const TreemapTreeTable();
-
-  @override
-  TreemapTreeTableState createState() => TreemapTreeTableState();
-}
-
-class TreemapTreeTableState extends State<TreemapTreeTable>
-    with AutoDisposeMixin {
-  TreeColumnData<TreemapNode> treeColumn =
-      _LibraryRefColumn(currentRootLevel: 0);
-  final List<ColumnData<TreemapNode>> columns = [];
-
-  CodeSizeController controller;
-
-  TreemapNode currentRoot;
-  @override
-  void initState() {
-    super.initState();
-    setupColumns();
-  }
-
-  void setupColumns() {
-    columns.addAll([
+class CodeSizeTable extends StatelessWidget {
+  factory CodeSizeTable({
+    @required rootNode,
+    @required totalSize,
```

Comment:
you don't need to pass in totalSize. You can access it directly in the factory via `rootNode.root.byteSize`

Reply:
Done